plugins/modules/ntnx_clusters.py

```diff
@@ -0,0 +1,522 @@
+                        username=dict(type="str"),
+                        password=dict(type="str", no_log=True),
+                    ),
+                ),
+                proxy_type_list=dict(type="list", elements="str"),
+            )
+        ),
+        server_type=dict(type="str"),
+    )
+
+    http_proxy_whitelist_spec = dict(
+        target=dict(type="str"),
+        target_type=dict(type="str"),
+    )
+
+    default_vswitch_config_spec = dict(
+        nic_teaming_policy=dict(type="str"),
+        uplink_grouping=dict(type="str"),
+    )
+
+    network_spec = dict(
+        external_ip=dict(type="str"),
+        fully_qualified_domain_name=dict(type="str"),
+        external_data_services_ip=dict(type="str"),
+        external_subnet=dict(type="str"),
+        internal_subnet=dict(type="str"),
+        masquerading_ip=dict(type="str"),
+        masquerading_port=dict(type="str"),
+        domain_server=dict(type="dict", options=domain_server_spec),
+        nfs_subnet_whitelist=dict(type="list", elements="str"),
+        name_server_ip_list=dict(type="list", elements="str"),
+        ntp_server_ip_list=dict(type="list", elements="str"),
+        http_proxy_list=dict(type="list", elements="dict", options=http_proxy_list_spec),
+        smtp_server=dict(type="dict", options=smtp_server_spec),
+        http_proxy_whitelist=dict(type="list", elements="dict", options=http_proxy_whitelist_spec),
+        default_vswitch_config=dict(type="dict", options=default_vswitch_config_spec),
+    )
+    authorized_public_key_list_spec = dict(
+        name=dict(type="str"),
+        key=dict(type="str", no_log=False),
+    )
+
+    module_args = dict(
+        cluster=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive, required=True,
+        ),
+        authorized_public_key_list=dict(type="list", elements="dict", options=authorized_public_key_list_spec),
+        timezone=dict(type="str"),
+        supported_information_verbosity=dict(type="str"),
+        redundancy_factor=dict(type="int"),
+        network=dict(type="dict", options=network_spec),
+    )
+
+    return module_args
+
+
+def update_cluster_config(module, result):
+    cluster = Cluster(module)
+    old_spec = cluster.get_current_spec()
+    spec, error = cluster.get_spec(old_spec)
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating cluster Spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+    uuid = spec["metadata"]["uuid"]
+    resp = cluster.update(spec, uuid)
+    result["task_uuid"] = resp["status"]["execution_context"]["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+        resp = cluster.read(uuid)
+        result["response"] = resp
+
+    # compare the specs before and after.  currently only way to determine if something changed
+    if old_spec["spec"] != result["response"]["spec"]:
```

Comment:
For idempotency checks, we compare old spec and update spec before making api call to update. If there is no difference then we skip the update call. This is how we make our modules idempotent. Please refer other modules for same. You can compare the specs directly if they dont required indepth checks. It should not make update call if old spec and update spec are equivalent. Ex: https://github.com/nutanix/nutanix.ansible/blob/b82b58d0a767e821f64107c2f363cebfaa892822/plugins/modules/ntnx_protection_rules.py#L700

Reply:
Post update call changed will be always true incase idempotency checks are placed before calls